.circleci/run_on_circle.js

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env node
+'use strict';
+
+/* eslint-disable camelcase */
+const http = require('https');
+const inquirer = require('inquirer');
+const simpleGit = require('simple-git')();
+
+// returns a Promise that resolves with branch information
+function getBranches() {
+  return new Promise((resolve, reject) => {
+    simpleGit.branchLocal((e, branches) => {
+      if (e) {
+        reject(e);
+      } else {
+        resolve(branches);
+      }
+    });
+  });
+}
+
+// generates a circleCI request using parameters provided
+function generateBuildRequest(program) {
+  // https://circleci.com/api/v2/project/github/twilio/twilio-video.js/pipeline
+  var options = {
+    hostname: 'circleci.com',
+    path: '/api/v2/project/github/twilio/twilio-video.js/pipeline',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': 'Basic ' + Buffer.from(program.token).toString('base64'),
+      'Accept': 'application/json',
+      'Host': 'circleci.com',
+    }
+  };
+
+  const body = {
+    branch: program.branch,
+    parameters: {
+      pr_workflow: program.workflow === 'pr',
+      custom_workflow: program.workflow === 'custom',
+      environment: program.environment,
+    }
+  };
+
+  if (program.workflow === 'custom') {
+    body.parameters.browser = program.browser;
+    body.parameters.bver = program.bver;
+    body.parameters.topology = program.topology;
+  }
+
+  return { options, body };
+}
+
+// sends a request using given options/body
+function triggerBuild({options, body}) {
+  console.log('options:', options);
+  console.log('body:', body);
+  return new Promise((resolve, reject) => {
```

Comment:
@makarandp0 I've seen this pattern a few times now, so maybe we can refactor the code for making Circle CI API requests into its own module?

Reply:
@manjeshbhargav - This `triggerBuild` method does not have anything circleCI specific -  it's just sending a request over.  CircleCI request specific logic is in `generateBuildRequest` - and that is tied closely to how we have structured our parameters in `config.yml`. For now this is the only place that does anything with circleCI request, I will look into making it reusable once we need it for release tool as well.